Added license column to dataset csv files

kcrans opened the pull request:
closes #1710
- Added a new "License" column to all documentation csv file.
- Used [SPDX license identifiers](https://spdx.org/licenses/) when possible.
- Datasets in the public domain I classified as "public domain." Copernicus data that uses there own licenses is called "CSCDA-ESA." OpenRAIL was not in the SPDX list, so I just used "OpenRAIL" when applicable.
- Null values of "-" where used in cases where licenses where missing or unclear.
- Swapped the NCCM and NLCD rows in geo_datasets.csv due to improper alphabetical ordering.
- Updated datasets.rst and models.rst to incorporate the correct number of columns for the csv tables in the documentation.

kcrans commented:
@microsoft-github-policy-service agree

adamjstewart commented on docs/api/landsat_pretrained_weights.csv:
These are all CC0-1.0: https://huggingface.co/torchgeo/ssl4eo_landsat

kcrans replied:
Got it, fixed.

adamjstewart commented:
A couple of general comments as I go through these:

* According to [SPDX](https://wiki.spdx.org/view/Legal_Team/Decisions/Dealing_with_Public_Domain_within_SPDX_Files), "public domain" isn't really specific and there is no exact license for it. So instead of CC0 we should just say "public domain" and let company lawyers decide if they're allowed to use it.
* I'm afraid to use "Assumed XYZ" because it could put us in legal trouble if we assumed incorrectly. So either say "public domain" if we know that's the case, or leave it blank until we know for certain.

adamjstewart commented on docs/api/geo_datasets.csv:
```diff
@@ -1,23 +1,23 @@
+`EDDMapS`_,Points,Citizen Scientists,"Assumed CC0-1.0",-,-
+`EnviroAtlas`_,"Imagery, Masks","NAIP, NLCD, OpenStreetMap","CC-BY-4.0",-,1
+`Esri2020`_,Masks,Sentinel-2,"CC-BY-4.0",-,10
+`EU-DEM`_,Masks,"Aster, SRTM, Russian Topomaps",-,-,25
```
They have their own [custom license](https://spacedata.copernicus.eu/documents/20123/121286/CSCDA_ESA_Mission-specific+Annex_31_Oct_22.pdf/fb109818-56ad-bbee-053c-d972aed25ce6?t=1674741175657), maybe call it CSCDA-ESA.

adamjstewart commented on docs/api/geo_datasets.csv:
```diff
@@ -1,23 +1,23 @@
+`EnviroAtlas`_,"Imagery, Masks","NAIP, NLCD, OpenStreetMap","CC-BY-4.0",-,1
+`Esri2020`_,Masks,Sentinel-2,"CC-BY-4.0",-,10
+`EU-DEM`_,Masks,"Aster, SRTM, Russian Topomaps",-,-,25
+`GBIF`_,Points,Citizen Scientists,-,-,-
```
According to https://www.gbif.org/terms it should be "CC0-1.0 OR CC-BY-4.0 OR CC-BY-NC-4.0".

kcrans replied:
Yep, because the license varies by datapoint I decided to not specify a specific license. But your suggestion is more clear, so I'll switch to that.

kcrans commented:
> A couple of general comments as I go through these:
> 
> * According to [SPDX](https://wiki.spdx.org/view/Legal_Team/Decisions/Dealing_with_Public_Domain_within_SPDX_Files), "public domain" isn't really specific and there is no exact license for it. So instead of CC0 we should just say "public domain" and let company lawyers decide if they're allowed to use it.
> * I'm afraid to use "Assumed XYZ" because it could put us in legal trouble if we assumed incorrectly. So either say "public domain" if we know that's the case, or leave it blank until we know for certain.

Ok, I removed all "Assumed XYZ" cases and added "public domain" as the license when appropriate.

adamjstewart reviewed (changes requested):
My brain hurts.

Thank you for going through so many datasets and weights. I only found a few licenses I think should be changed.

P.S. Technically, only CSV entries containing a comma need to be in quotes. Also, would be nice to keep the EOL char at the end of each file.

kcrans commented on docs/api/non_geo_datasets.csv:
```diff
@@ -1,51 +1,51 @@
+`SustainBench Crop Yield`_,R,MODIS,"CC-BY-SA-4.0",11k,-,32x32,-,MSI
+`Tropical Cyclone`_,R,GOES 8--16,"CC-BY-4.0","108,110",-,256x256,4K--8K,MSI
+`UC Merced`_,C,USGS National Map,-,"2,100",21,256x256,0.3,RGB
+`USAVars`_,R,NAIP Aerial,"CC-BY-4.0",100K,-,-,4,"RGB, NIR"
```
I found it for the original dataset here: https://codeocean.com/capsule/6456296/tree/v2/data/LICENSE.

adamjstewart replied:
Sounds good, @calebrob6 I'm going to update the license on our HF copy to match if that's alright with you.

kcrans commented on docs/api/non_geo_datasets.csv:
```diff
@@ -1,51 +1,51 @@
+`ChaBuD`_,CD,Sentinel-2,"OpenRAIL",356,2,512x512,10,MSI
+`Cloud Cover Detection`_,S,Sentinel-2,"CC-BY-4.0","22,728",2,512x512,10,MSI
+`COWC`_,"C, R","CSUAV AFRL, ISPRS, LINZ, AGRC","AGPL-3.0-only","388,435",2,256x256,0.15,RGB
+`Kenya Crop Type`_,S,Sentinel-2,"Apache-2.0","4,688",7,"3,035x2,016",10,MSI
```
Oops I think I must have confused the license for the solutions with the dataset license. It looks like the license is actually CC BY-SA 4.0 based off the "Rules" section on the competition [site.](https://zindi.africa/competitions/iclr-workshop-challenge-2-radiant-earth-computer-vision-for-crop-recognition)

adamjstewart replied:
Looks right to me.

adamjstewart replied:
```suggestion
`Kenya Crop Type`_,S,Sentinel-2,"CC-BY-SA-4.0","4,688",7,"3,035x2,016",10,MSI
```

adamjstewart commented on docs/api/geo_datasets.csv:
```diff
@@ -1,23 +1,23 @@
+`LandCover.ai Geo`_,"Imagery, Masks",Aerial,"CC-BY-NC-SA-4.0","4,200--9,500",0.25--0.5
+`Landsat`_,Imagery,Landsat,"public domain","8,900x8,900",30
+`NAIP`_,Imagery,Aerial,"public domain","6,100x7,600",1
+`NCCM`_,Masks,Sentinel-2,"CC-BY-4.0",-,10
```
According to https://figshare.com/articles/figure/The_10-m_crop_type_maps_in_Northeast_China_during_2017-2019/13090442 it's CC-BY-4.0, so this is correct.

kcrans commented on docs/api/geo_datasets.csv:
```diff
@@ -1,23 +1,23 @@
+`Canadian Building Footprints`_,Geometries,Bing Imagery,"ODbL-1.0",-,-
+`Chesapeake Land Cover`_,"Imagery, Masks",NAIP,-,-,1
+`Global Mangrove Distribution`_,Masks,"Remote Sensing, In Situ Measurements","public domain",-,3
+`Cropland Data Layer`_,Masks,Landsat,"CC0-1.0",-,30
```
I got it from here: https://catalog.data.gov/dataset/cropscape-cropland-data-layer but it actually looks like it is public domain and data.gov just assigned it CC0. We should given precedence to the USDA's license description.

kcrans commented on docs/api/non_geo_datasets.csv:
```diff
@@ -1,51 +1,51 @@
+`LandCover.ai`_,S,Aerial,"CC-BY-NC-SA-4.0","10,674",5,512x512,0.25--0.5,RGB
+`LEVIR-CD+`_,CD,Google Earth,-,985,2,"1,024x1,024",0.5,RGB
+`LoveDA`_,S,Google Earth,"CC-BY-NC-SA-4.0","5,987",7,"1,024x1,024",0.3,RGB
+`MapInWild`_,S,"Sentinel-1/2, ESA WorldCover, NOAA VIIRS DNB","CC-BY-4.0",1018,1,1920x1920,10--463.83,"SAR, MSI, 2020_Map, avg_rad"
```
I found the CC-BY-4.0 license from here: https://dataverse.harvard.edu/dataverse/mapinwild which I think is the original complete dataset?

kcrans commented on docs/api/non_geo_datasets.csv:
```diff
@@ -1,51 +1,51 @@
+`Potsdam`_,S,Aerial,-,38,6,"6,000x6,000",0.05,MSI
+`ReforesTree`_,"OD, R",Aerial,-,100,6,"4,000x4,000",0.02,RGB
+`RESISC45`_,C,Google Earth,-,"31,500",45,256x256,0.2--30,RGB
+`Rwanda Field Boundary`_,S,Planetscope,-,70,2,256x256,4.7,RGB + NIR
```
Oops, yeah I think I gave that one a null value back when I was classifying all datasets with multiple licenses as null.

kcrans commented on docs/api/non_geo_datasets.csv:
```diff
@@ -1,51 +1,51 @@
+`SEN12MS`_,S,"Sentinel-1/2, MODIS","CC-BY-4.0","180,662",33,256x256,10,"SAR, MSI"
+`SKIPP'D`_,R,"Fish-eye","CC-BY-4.0","363,375",-,64x64,-,RGB
+`So2Sat`_,C,Sentinel-1/2,"CC-BY-4.0","400,673",17,32x32,10,"SAR, MSI"
+`SpaceNet`_,I,WorldView-2/3 Planet Lab Dove,-,"1,889--28,728",2,102--900,0.5--4,MSI
```
Ok, thanks for digging deeper into that.

kcrans commented on docs/api/non_geo_datasets.csv:
```diff
@@ -1,51 +1,51 @@
+`Kenya Crop Type`_,S,Sentinel-2,"Apache-2.0","4,688",7,"3,035x2,016",10,MSI
+`DeepGlobe Land Cover`_,S,DigitalGlobe +Vivid,-,803,7,"2,448x2,448",0.5,RGB
+`DFC2022`_,S,Aerial,"CC-BY-4.0","3,981",15,"2,000x2,000",0.5,RGB
+`ETCI2021 Flood Detection`_,S,Sentinel-1,-,"66,810",2,256x256,5--20,SAR
```
I'm not sure about that one as the competition website says this: https://competitions.codalab.org/competitions/30440#learn_the_details-terms_and_conditions. I think someone unrelated might just have uploaded the dataset to Kaggle?

kcrans commented on docs/api/non_geo_datasets.csv:
```diff
@@ -1,51 +1,51 @@
+`MapInWild`_,S,"Sentinel-1/2, ESA WorldCover, NOAA VIIRS DNB","CC-BY-4.0",1018,1,1920x1920,10--463.83,"SAR, MSI, 2020_Map, avg_rad"
+`Million-AID`_,C,Google Earth,-,1M,51--73,,0.5--153,RGB
+`NASA Marine Debris`_,OD,PlanetScope,"Apache-2.0",707,1,256x256,3,RGB
+`OSCD`_,CD,Sentinel-2,"CC-BY-4.0",24,2,"40--1,180",60,MSI
```
I found it here: https://ieee-dataport.org/open-access/oscd-onera-satellite-change-detection.

adamjstewart reviewed (approved):
This list is accurate to the best of my abilities, but I'm not a lawyer. If any dataset or model authors see this and find any mistakes or change their licenses, please submit an issue or PR to update this list.

adamjstewart commented:
Thanks for the contribution @kcrans!

lukasValentin commented on docs/api/non_geo_datasets.csv:
```diff
@@ -1,51 +1,51 @@
+`VHR-10`_,I,"Google Earth, Vaihingen",-,800,10,"358--1,728",0.08--2,RGB
+`Western USA Live Fuel Moisture`_,R,"Landsat8, Sentinel-1","CC-BY-NC-ND-4.0",2615,-,-,-,-
+`xView2`_,CD,Maxar,"CC-BY-NC-SA-4.0","3,732",4,"1,024x1,024",0.8,RGB
+`ZueriCrop`_,"I, T",Sentinel-2,-,116K,48,24x24,10,MSI
```
Hi @adamjstewart, sorry for my late response - I missed your question. In case your question is still relevant I recommend to check with @0zgur0 who should know more about ZueriCrops as I also could not find if there's a license and what kind of
